### PR TITLE
FIX: [imx] dynamic allocation of buffer array

### PR DESCRIPTION
**Pull request by koying**
This does dynamic allocation of the vpu buffers.
It solves crashes I had because 20 was not enough.

### CODE REVIEW

File: xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.h [hunk 1]
```diff
@@ -32,20 +32,24 @@
 //#define TRACE_FRAMES
 
 // FIXME TODO Develop real proper CVPUBuffer class
```
Comment (wolfgar) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.h]: You can remove this comment now... ;)

**Comment by wolfgar** [comment]
I never experienced the crash you mention but it obviously the proper way of doing it...
Ironically, this bad 20 max constant value comes from the official freescale example in libfslvpuwrap-1.0.35 package (file decode_stream.c)

**Comment by koying** [comment]
Don't forget we are allocating frame buffers this way, too, now,  while it
is allocated on v4l in the examples (I assume)

**Comment by wolfgar** [comment]
Very true ! Thanks for pointing this...

**Pull request merged**